Marlin/UBL_G29.cpp

```diff
@@ -712,58 +712,193 @@
+      DEPLOY_PROBE();
+
+// this is a copy of the G29 AUTO_BED_LEVELING_BILINEAR method/code
+      #undef PROBE_Y_FIRST
+      #if ENABLED(PROBE_Y_FIRST)
+        #define PR_OUTER_VAR xCount
+        #define PR_OUTER_NUM UBL_MESH_NUM_X_POINTS
+        #define PR_INNER_VAR yCount
+        #define PR_INNER_NUM UBL_MESH_NUM_Y_POINTS
+      #else
+        #define PR_OUTER_VAR yCount
+        #define PR_OUTER_NUM UBL_MESH_NUM_Y_POINTS
+        #define PR_INNER_VAR xCount
+        #define PR_INNER_NUM UBL_MESH_NUM_X_POINTS
+      #endif
+
+      bool zig = PR_OUTER_NUM & 1;  // Always end at RIGHT and BACK_PROBE_BED_POSITION
+
+      // Outer loop is Y with PROBE_Y_FIRST disabled
+      for (uint8_t PR_OUTER_VAR = 0; PR_OUTER_VAR < PR_OUTER_NUM; PR_OUTER_VAR++) {
+
+        int8_t inStart, inStop, inInc;
+
+        if (zig) { // away from origin
+          inStart = 0;
+          inStop = PR_INNER_NUM;
+          inInc = 1;
+        }
+        else {     // towards origin
+          inStart = PR_INNER_NUM - 1;
+          inStop = -1;
+          inInc = -1;
+        }
+
+        zig = !zig; // zag
+
+        // Inner loop is Y with PROBE_Y_FIRST enabled
+        for (int8_t PR_INNER_VAR = inStart; PR_INNER_VAR != inStop; PR_INNER_VAR += inInc) {
+// end of G29 AUTO_BED_LEVELING_BILINEAR method/code
+          if (ubl_lcd_clicked()) {
+            SERIAL_PROTOCOLLNPGM("\nMesh only partially populated.\n");
+            lcd_quick_feedback();
+            STOW_PROBE();
+            while (ubl_lcd_clicked() ) {
+              idle();
+            }
+            ubl_has_control_of_lcd_panel = false;
+            restore_ubl_active_state_and_leave();
+            safe_delay(50);  // Debounce the Encoder wheel
+            return;
+          }
+
+          const float xNozzle = ubl.map_x_index_to_bed_location(xCount),
+                      yNozzle = ubl.map_y_index_to_bed_location(yCount);
+
+          bool Probe_can_reach = position_is_reachable_UBL(xNozzle, yNozzle, PROBE_CHECK);
+          bool Within_bed = position_is_reachable_UBL(xNozzle, yNozzle, !PROBE_CHECK);
+
+          if (Probe_can_reach && Within_bed && isnan(z_values[xCount][yCount])) {
+
+            const float measured_z = probe_pt(xNozzle - X_PROBE_OFFSET_FROM_EXTRUDER, yNozzle - Y_PROBE_OFFSET_FROM_EXTRUDER, stow_probe, g29_verbose_level);
+            z_values[xCount][yCount] = measured_z + zprobe_zoffset;
+
+            if (do_ubl_mesh_map) ubl.display_map(map_type);
+          }
+//          else if (!Within_bed) z_values[xCount][yCount] = ??   //something to flag that this point can't be used
+        }
+      }
+      STOW_PROBE();
+      restore_ubl_active_state_and_leave();
+
+      do_blocking_move_to_xy(
+        constrain(lx - (X_PROBE_OFFSET_FROM_EXTRUDER), X_MIN_POS, X_MAX_POS),
+        constrain(ly - (Y_PROBE_OFFSET_FROM_EXTRUDER), Y_MIN_POS, Y_MAX_POS)
+        );

```

Comment:
Note that this doesn't work for SCARA. The XY offsets must be rotated by the sum of the arm angles.